bin/tools.py

```diff
@@ -861,6 +876,36 @@ def run_parsed_arguments(args):
         sys.exit(1)
 
 
+def read_personal_config():
+    args = {}
+
+    if is_windows():
+        home_config = Path(os.getenv('AppData'))
+    else:
+        home_config = (
+            Path(os.getenv('XDG_CONFIG_HOME'))
+            if os.getenv('XDG_CONFIG_HOME')
+            else Path(os.getenv('HOME')) / '.config'
+        )
+
+    for config_file in [
+        # Highest prio: contest directory
+        Path() / '.bapctools.yaml',
+        Path() / '..' / '.bapctools.yaml',
+        # Lowest prio: user config directory
+        home_config / 'bapctools' / 'config.yaml',
+    ]:
+        if not config_file.is_file():
+            continue
+        config_data = read_yaml(config_file)
+        assert isinstance(config_data, dict)
```

Comment:
maybe allow `None` as well for empty config
```suggestion
        assert isinstance(config_data, [dict, None])
```

Reply:
or add `or {}` to the line above

Reply:
I went for the second option :slightly_smiling_face: (57be5a8)